src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadMetadata.java

```diff
@@ -4,54 +4,82 @@
-        final List<SAMSequenceRecord> contigs = header.getSequenceDictionary().getSequences();
-        if ( contigs.size() > Short.MAX_VALUE ) throw new GATKException("Too many reference contigs.");
-        contigNameToID = new HashMap<>(SVUtils.hashMapCapacity(contigs.size()));
-        final int nContigs = contigs.size();
-        for ( int contigID = 0; contigID < nContigs; ++contigID ) {
-            contigNameToID.put(contigs.get(contigID).getSequenceName(), (short)contigID);
+                         final JavaRDD<GATKRead> reads ) {
+        contigNameToID = buildContigNameToIDMap(header);
+
+        final int nReadGroups = header.getReadGroups().size();
+        final List<PartitionStatistics> perPartitionStatistics =
+                reads.mapPartitions(readItr -> Collections.singletonList(new PartitionStatistics(readItr, nReadGroups)))
+                        .collect();
+        nPartitions = perPartitionStatistics.size();
+        nReads = perPartitionStatistics.stream().mapToLong(PartitionStatistics::getNReads).sum();
+        maxReadsInPartition = perPartitionStatistics.stream().mapToLong(PartitionStatistics::getNReads).max().orElse(0L);
+        final long nReadBases = perPartitionStatistics.stream().mapToLong(PartitionStatistics::getNBases).sum();
+        final long nRefBases = header.getSequenceDictionary().getSequences()
+                .stream().mapToLong(SAMSequenceRecord::getSequenceLength).sum();
+        coverage = (int)((nReadBases + nRefBases - 1) / nRefBases);
+        readGroupToFragmentStatistics = new HashMap<>(SVUtils.hashMapCapacity(header.getReadGroups().size()));
+        final Map<String, long[]> combinedMaps =
+                perPartitionStatistics.stream()
+                        .map(PartitionStatistics::getReadGroupToFragmentSizeCountMap)
+                        .reduce(new HashMap<>(SVUtils.hashMapCapacity(nReadGroups)), ReadMetadata::combineMaps);
+        for ( final Map.Entry<String, long[]> entry : combinedMaps.entrySet() ) {
+            readGroupToFragmentStatistics.put(entry.getKey(),
+                    new ReadGroupFragmentStatistics(entry.getValue()));
         }
+    }
 
-        final List<SAMReadGroupRecord> readGroups = header.getReadGroups();
-        if ( readGroups.size() != statistics.size() ) throw new GATKException("Wrong number of statistics for read groups.");
-        readGroupToFragmentStatistics = new HashMap<>(SVUtils.hashMapCapacity(readGroups.size()));
-        final int nReadGroups = readGroups.size();
-        for ( int readGroupId = 0; readGroupId < nReadGroups; ++readGroupId ) {
-            readGroupToFragmentStatistics.put(readGroups.get(readGroupId).getId(), statistics.get(readGroupId));
+    @VisibleForTesting
+    ReadMetadata( final SAMFileHeader header, final ReadGroupFragmentStatistics stats ) {
+        contigNameToID = buildContigNameToIDMap(header);
+        nPartitions = 1;
+        nReads = 2;
```

Comment:
Just made-up stuff for testing.  I'll add args to the constructor to cover these properties, and then the magic numbers will correspond to test data properties in the testing class and will disappear from here.